flake.nix

```diff
@@ -185,7 +185,7 @@
           buildInputs =
             [ rust ]
             ++ missingSysPkgs
-            ++ (if !isDevShell then [ cargoHome ] else [ ]);
+            ++ (if !isDevShell then [ cargoHome ] else [ pkgs.rust-analyzer ]);
```

Comment:
I know my question is unrelated to your change, but while we're reviewing this line:
Why do we add `cargoHome` only if not in a dev shell? What happens if we are in a dev shell?

No matter the answer to that question, can we please add a comment to explain that to future readers? :pray:

Reply:
Let's do this in a separate PR.